borsh-ts/test/serialize.test.js

```diff
@@ -158,3 +158,26 @@ test('base encode and decode test', async () => {
     const value = '244ZQ9cgj3CQ6bWBdytfrJMuMQ1jdXLFGnr4HhvtCTnM';
     expect(borsh.baseEncode(borsh.baseDecode(value))).toEqual(value);
 });
+
+test('serialize with custom writer/reader', async () => {
+    class ExtendedWriter extends borsh.BinaryWriter {
+        writeDate(value) {
+            this.writeU64(value.getTime());
+        }
+    }
+
+    class ExtendedReader extends borsh.BinaryReader {
+        readDate() {
+            const value = this.readU64();
+            return new Date(value.toNumber());
+        }
+    }
+
+    const time = 'Aug 12, 2021 12:00:00 UTC+00:00';
+    const value = new Test({ x: new Date(time) });
+    const schema = new Map([[Test, {kind: 'struct', fields: [['x', 'date']]}]]);
```

Comment:
I think it will but just to make sure: if there's another field that's not Date, the customized writer should still works? To test the composable of the customized writer/reader.

Reply:
Yes, as long as the custom writer inherits from the original